qiskit/primitives/backend_sampler_v2.py

```diff
@@ -62,6 +68,16 @@ class _MeasureInfo:
     start: int
 
 
+ResultMemory = Union[list[str], list[list[float]], list[list[list[float]]]]
```

Comment:
I don't think you need to import `Union`.
```suggestion
ResultMemory = list[str] | list[list[float]] | list[list[list[float]]]
```

Reply:
I wrote it this way at first, but it raises a `TypeError` exception on Python 3.9. The type union feature was added in [Python 3.10](https://peps.python.org/pep-0604/). This syntax can be put in type signatures on 3.9 if `from __future__ import annotations` is used to skip type signature evaluation but the usage here is not inside of a type signature.